src/Workspaces/CSharp/Portable/CodeGeneration/NamedTypeGenerator.cs

```diff
@@ -196,7 +198,14 @@ private static MemberDeclarationSyntax RemoveAllMembers(MemberDeclarationSyntax
             TypeDeclarationSyntax typeDeclaration;
             if (namedType.IsRecord)
             {
-                typeDeclaration = SyntaxFactory.RecordDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken());
+                if (namedType.IsReferenceType)
+                {
+                    typeDeclaration = SyntaxFactory.RecordDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken());
+                }
+                else
+                {
+                    typeDeclaration = SyntaxFactory.RecordStructDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken());
```

Comment:
:memo: not tested (we don't yet have scenarios to generate record structs, but implementation is straightforward

Reply:
i think we'll be fine.  i'm going to write a feature to covert a tuple to a named record.  So i'll test it that way.